### PR TITLE
Merge dev branch into master

### PR DESCRIPTION
**Pull request by rqrauhvmra**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,7 +162,29 @@ or a button with the class name `lightbox` and a `data-target` attribute:
...
+<button type="button" data-type="youtube" data-id="KU2sSZ_90PY" class="lightbox">
+  Open YouTube video
+</button>
+```
+
+In any case, the attribute `data-type` with the value `youtube` must be added.
+
+#### Optional attributes
+
+`data-controls` indicates whether the video player controls are displayed: `0` do not display and `1` display controls in the player. You can also set the height with `data-height` and the width of the player with `data-width` but I recommend to use a external library for responsive iframes.
```
Comment (midzer) [inline README.md]: How about doing responsive `iframes` with (copied) CSS classes from http://getbootstrap.com/docs/4.1/utilities/embed/ ?

Comment (rqrauhvmra) [reply README.md]: I'll take a look.

**Comment by midzer** [comment]
Would be awesome if you might sum up important changes from time to time, @rqrauhvmra
This way, I can better help to double-check Tobi for all components to be still intact before 1.8.0 rls is coming up :)

**Comment by rqrauhvmra** [comment]
> 
> 
> Would be awesome if you might sum up important changes from time to time, @rqrauhvmra
> This way, I can better help to double-check Tobi for all components to be still intact before 1.8.0 rls is coming up :)

You are absolutely right @midzer. I promise improvement.

Until now, I revert some changes and improved some things. I will explain my changes in detail in the future.

**Comment by midzer** [comment]
Hi again,
I had some time to test against latest development:

Looks like next/prev swipe is currently broken for videos on mobile. This used to be working, so probably minor regression here. Mysterycally vertical close swipe is working fine.

Would be cool if you could help on this issue, as I'm more or less on vacation right now. You can check https://feuerwehr-eisolzried.de/media/ for an example gallery

**Comment by rqrauhvmra** [comment]
> Looks like next/prev swipe is currently broken for videos on mobile. This used to be working, so probably minor regression here. Mysterycally vertical close swipe is working fine.
> 
> Would be cool if you could help on this issue, as I'm more or less on vacation right now. You can check https://feuerwehr-eisolzried.de/media/ for an example gallery

Hi @midzer 

The falling snowflakes eat a lot of CPU on my old Android 🙈 but the next/prev swipe navigation looks good for me (for the images under the "Bilder" and for the videos under the "Video" tab) @ Firefox 63.0.2 on Android 4.4.2.

**Comment by midzer** [comment]
Thanks for your response.
Yeah, those snowflakes are a heavy drainer. Maybe I should wait until there's in fact snow lying around in my area :-)

Still can't swipe next video for those videos using same Firefox version like you and latest Chromium 72 on mobile. Dunno whats going on there...I hope some1 else reading this thread tests again dev brach, too

**Comment by rqrauhvmra** [comment]
> Still can't swipe next video for those videos using same Firefox version like you and latest Chromium 72 on mobile. Dunno whats going on there...I hope some1 else reading this thread tests again dev brach, too

Hmm... 🤔

You are not trying to swipe directly on the video? The area is excluded from swiping (due to the video controls). I swipe at the bottom.

Does it work without the snow? I would like to exclude that. (but it works also with it for me)

**Comment by midzer** [comment]
> You are not trying to swipe directly on the video? The area is excluded from swiping (due to the video controls). I swipe at the bottom.

Ahhhh, now it's working!!
But why exclude video frame for swiping. I don't understand. It used to work perfectly (at least for me). Those HTML5 video controls don't stand in swiping's way.

> Does it work without the snow? I would like to exclude that. (but it works also with it for me)

Yup, on next update I'm gonna remove snow again and wait for new real life snow to bring it back :-D

**Comment by rqrauhvmra** [comment]
> 
> 
> > You are not trying to swipe directly on the video? The area is excluded from swiping (due to the video controls). I swipe at the bottom.
> 
> Ahhhh, now it's working!!
> But why exclude video frame for swiping. I don't understand. It used to work perfectly (at least for me). Those HTML5 video controls don't stand in swiping's way.

I had some issues. The video started playing for example (no autoplay). I will est that again and make the conditions in the [function isIgnoreElement](https://github.com/rqrauhvmra/Tobi/blob/64e5924a9ef6d5c51ad50274069484f585a28b01/js/tobi.js#L1269) optionable.

**Comment by rqrauhvmra** [comment]
> Still looking forward to some1 leaving review or feedback in this (big) thread to have this one merged before Christmas :)

I hope it will not be so long 😃 I will just adapt and test a few little things. I've just become a father and have too much work but there are only a few days left until I release 1.8.0. 🎉

**Comment by midzer** [comment]
Sorry, I did not want to put any pressure on release date. Real life always goes first. I can wait.

Happy fathers time :)

**Comment by viliusle** [comment]
I tried to look at it, but its hard to review/understand all these code changes, there are lots of them.
So i tried to check if demo is working, but could not find any demo or tests. I used my old demo.html, but was not working, because of tag and logic changes on HTML.  I think we need something like that in each branch, that should be supported and easily accessed.

Another problem: tobi.js file is getting more and more content, now it has 1300 lines, and its too much code in 1 file. And it will become bigger in realease. We should think about splitting logic into several files. It should not be a problem for end user, since he would use minified version. Of course code splitting  requires some build process, like webpack or bower (obsolete, but still good for small projects).

So in general, please think about including tests/demo in code branch and splitting code in future versions.

**Comment by viliusle** [comment]
I can approve, but still waiting comments about issues above. (or we can move it to separate issues for discussion).

**Comment by rqrauhvmra** [comment]
Hi @viliusle 

you're absolutely right. I'll include an demo template. I also thought about code splitting (rewrite _Tobi_ in ES6 and use modules).

I open an issue for that.

**Pull request merged**